app/services/ZuoraService.scala

```diff
@@ -17,10 +17,33 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
+sealed trait ZuoraFilter {
+  def toFilterString: String
+}
+
+case class SimpleFilter(key: String, value: String) extends ZuoraFilter {
+  override def toFilterString = s"$key='$value'"
+}
+
+case class AndFilter(clauses: (String, String)*) extends ZuoraFilter {
+  override def toFilterString =
+    clauses.map(p => SimpleFilter.tupled(p).toFilterString).mkString(" AND ")
+}
+
+case class OrFilter(clauses: (String, String)*) extends ZuoraFilter {
```

Comment:
Was just wondering if maybe this should just take a `Seq`, as it mainly seems to be called with Seq and ':_*'?